tests/e2e.sh

```diff
@@ -97,24 +98,45 @@ function kube_state_metrics_up() {
 function test_daemonset() {
     sed -i "s|${KUBE_STATE_METRICS_CURRENT_IMAGE_NAME}:v.*|${KUBE_STATE_METRICS_IMAGE_NAME}:${KUBE_STATE_METRICS_IMAGE_TAG}|g" ./examples/daemonsetsharding/deployment.yaml
     sed -i "s|${KUBE_STATE_METRICS_CURRENT_IMAGE_NAME}:v.*|${KUBE_STATE_METRICS_IMAGE_NAME}:${KUBE_STATE_METRICS_IMAGE_TAG}|g" ./examples/daemonsetsharding/daemonset.yaml
+    sed -i "s|${KUBE_STATE_METRICS_CURRENT_IMAGE_NAME}:v.*|${KUBE_STATE_METRICS_IMAGE_NAME}:${KUBE_STATE_METRICS_IMAGE_TAG}|g" ./examples/daemonsetsharding/deployment-no-node-pods.yaml
 
     kubectl get deployment -n kube-system
     kubectl create -f ./examples/daemonsetsharding
     ls ./examples/daemonsetsharding
     kube_state_metrics_up
-    curl -s "http://localhost:8001/api/v1/namespaces/kube-system/services/kube-state-metrics-shard:http-metrics/proxy/metrics" >${KUBE_STATE_METRICS_LOG_DIR}/pod-metrics
-    curl -s "http://localhost:8001/api/v1/namespaces/kube-system/services/kube-state-metrics:http-metrics/proxy/metrics" >${KUBE_STATE_METRICS_LOG_DIR}/nonpod-metrics
-    m1="$(cat ${KUBE_STATE_METRICS_LOG_DIR}/pod-metrics | grep "# TYPE kube_pod_info" || true)"
-    m2="$(cat ${KUBE_STATE_METRICS_LOG_DIR}/nonpod-metrics | grep "# TYPE kube_pod_info" || true)"
-    if [[ -z "${m1}" ]]; then
-        echo "can't found metric kube_pod_info from pod metrics service"
+    kubectl apply -f ./tests/e2e/testdata/pods.yaml
+    sleep 10
+    kubectl get deployment -n default
+    curl -s "http://localhost:8001/api/v1/namespaces/kube-system/services/kube-state-metrics-shard:http-metrics/proxy/metrics" >${KUBE_STATE_METRICS_LOG_DIR}/daemonset-scraped-metrics
+    curl -s "http://localhost:8001/api/v1/namespaces/kube-system/services/kube-state-metrics:http-metrics/proxy/metrics" >${KUBE_STATE_METRICS_LOG_DIR}/deployment-scraped-metrics
+    curl -s "http://localhost:8001/api/v1/namespaces/kube-system/services/kube-state-metrics-no-node-pods:http-metrics/proxy/metrics" >${KUBE_STATE_METRICS_LOG_DIR}/deployment-scraped-no-node-metrics
+
+    cat ${KUBE_STATE_METRICS_LOG_DIR}/daemonset-scraped-metrics >> ${KUBE_STATE_METRICS_LOG_DIR}/all-pod-metrics
+    cat ${KUBE_STATE_METRICS_LOG_DIR}/deployment-scraped-metrics >> ${KUBE_STATE_METRICS_LOG_DIR}/all-pod-metrics
+    cat ${KUBE_STATE_METRICS_LOG_DIR}/deployment-scraped-no-node-metrics >> ${KUBE_STATE_METRICS_LOG_DIR}/all-pod-metrics
```

Comment:
```suggestion
    cat ${KUBE_STATE_METRICS_LOG_DIR}/daemonset-scraped-metrics ${KUBE_STATE_METRICS_LOG_DIR}/deployment-scraped-metrics ${KUBE_STATE_METRICS_LOG_DIR}/deployment-scraped-no-node-metrics >> ${KUBE_STATE_METRICS_LOG_DIR}/all-pod-metrics
```

Fixes:
```
In ./tests/e2e.sh line 114:
    cat ${KUBE_STATE_METRICS_LOG_DIR}/daemonset-scraped-metrics >> ${KUBE_STATE_METRICS_LOG_DIR}/all-pod-metrics
    ^-- SC21[29](https://github.com/kubernetes/kube-state-metrics/actions/runs/9390258474/job/25859669104?pr=2388#step:5:30) (style): Consider using { cmd1; cmd2; } >> file instead of individual redirects.

For more information:
  https://www.shellcheck.net/wiki/SC2129 -- Consider using { cmd1; cmd2; } >>...In ./tests/e2e.sh line 114:
    cat ${KUBE_STATE_METRICS_LOG_DIR}/daemonset-scraped-metrics >> ${KUBE_STATE_METRICS_LOG_DIR}/all-pod-metrics
    ^-- SC21[29](https://github.com/kubernetes/kube-state-metrics/actions/runs/9390258474/job/25859669104?pr=2388#step:5:30) (style): Consider using { cmd1; cmd2; } >> file instead of individual redirects.

For more information:
  https://www.shellcheck.net/wiki/SC2129 -- Consider using { cmd1; cmd2; } >>...
```

https://github.com/kubernetes/kube-state-metrics/actions/runs/9390258474/job/25859669104?pr=2388#step:5:28